1.MLP_Binary_Classifier/MLP_utils/utils.py

```diff
@@ -0,0 +1,627 @@
+        )
+    return train_loss, train_acc, valid_loss, valid_acc, epochs_ran, model
+
+
+# Plot training data over epochs
+def plot_metric_vs_epoch(df, x, y1, y2):
+
+    sns.lineplot(x=x, y=y1, data=df)
+    sns.lineplot(x=x, y=y2, data=df)
+
+
+def test_optimized_model(model, test_loader, in_features, parameter_dict, params):
+    """
+    This function tests the trained optimized model on test dataset
+
+    Parameters:
+        None
+    Return:
+        y_pred_list : list
+            lsit of predicted values
+        y_pred_prob_list : list
+            lsit of probabilities for predicted values
+    """
+
+    model = optimized_model(in_features, parameter_dict)
+    model = model.to(params.DEVICE)
+    criterion = nn.BCEWithLogitsLoss()
+    optim_method = parameter_dict["optimizer"].strip("'")
+    optimizer = (
+        f'optim.{optim_method}(model.parameters(), lr=parameter_dict["learning_rate"])'
+    )
+    optimizer = eval(optimizer)
+
+    y_pred_prob_list = []
+    y_pred_list = []
+
+    # Loading the best model
+    # model.load_state_dict(torch.load('./state_dict.pt'))
+
+    with torch.no_grad():
+        model.eval()
+        for batch_idx, (X_test_batch, y_test_batch) in enumerate(test_loader):
+            X_test_batch = X_test_batch.to(params.DEVICE)
+            # PREDICTION
+            output = model(X_test_batch)
+            y_pred_prob = torch.sigmoid(output)
+            y_pred_prob_list.append(y_pred_prob.cpu().numpy())
+            y_pred = torch.round(y_pred_prob)
+            y_pred_list.append(y_pred.cpu().numpy())
+    y_pred_prob_list = [a.squeeze().tolist() for a in y_pred_prob_list]
+    y_pred_list = [a.squeeze().tolist() for a in y_pred_list]
+
+    return y_pred_list, y_pred_prob_list
+
+
+# If output list is nested
+def un_nest(lst):
+    """
+    returns an un-nested list from a nested list
+
+    Parameters:
+        lst : list
+            a list of lists
+    """
+    new_lst = []
+    for i in lst:
+        for j in i:
+            new_lst.append(j)
+    return new_lst
+
+
+def results_output(prediction_list, prediction_probability_list, test_data):
+    """
+    Function outputs visulaization of testing the model
+
+    Parameters:
+        prediction_list : lsit of predicted values from model
+        prediction_probability_list : list of probabailities of predicted values from model
+        test_data : input data to model
+
+    Return:
+        classification report
+        confusion matrix
+        AUC graph of accuracy and false positive rates
+
+    """
+    # Classification report
+    print(classification_report(test_data, prediction_list))
+
+    # confusion matrix
+    confusion_matrix(test_data, prediction_list)
+
+    # AUC graph of accuracy and false positive rates
+    plt.figure(figsize=(5.5, 4))
+    fpr, tpr, _ = roc_curve(test_data, prediction_probability_list)
```

Comment:
yayy! Love to see this. I would also consider adding precision_recall (but maybe beyond scope of this PR)

Reply:
I will add this to an issue for a later PR